### PR TITLE
[Drooble] Add extractor

### LINKED ISSUE #1527: Drooble
### Checklist

- [X] I'm reporting a new site support request
- [X] I've verified that I'm running yt-dlp version **2021.10.22**. ([update instructions](https://github.com/yt-dlp/yt-dlp#update))
- [X] I've checked that all provided URLs are alive and playable in a browser
- [X] I've checked that none of provided URLs [violate any copyrights](https://github.com/ytdl-org/youtube-dl#can-you-add-support-for-this-anime-video-site-or-site-which-shows-current-movies-for-free) or contain any [DRM](https://en.wikipedia.org/wiki/Digital_rights_management) to the best of my knowledge
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues including closed ones. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [ ] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and am willing to share it if required

### Region

Ukraine

### Example URLs

- Single audio: https://drooble.com/song/2858030
  - RAW (.mp3): `<audio preload="none" src="https://audios.drooblecdn.com/95894/audio/5882d174b49f605a9daa6b10d39d6a12"></audio>`
- Audio Album: https://drooble.com/karl340758/music/albums/2858031
- Audio Channel: https://drooble.com/karl340758/music

- Single video: https://drooble.com/karl340758/videos/2859183 
  > NOTE: Single video page has embedded YT video.
- Video Channel: https://drooble.com/karl340758/videos

### Description

`Drooble` is the social network for musicians.

### Verbose log

```shell
$ yt-dlp -Uv -F https://drooble.com/song/2858030
[debug] Command-line config: ['-Uv', '-F', 'https://drooble.com/song/2858030']
[debug] Encodings: locale UTF-8, fs utf-8, out UTF-8, err UTF-8, pref UTF-8[debug] yt-dlp version 2021.10.22 (zip)
[debug] Python version 3.7.3 (CPython 64bit) - Linux-4.19.0-18-amd64-x86_64-with-MX-19.4-patito_feo
[debug] exe versions: none
[debug] Optional libraries: sqlite
[debug] Proxy map: {}
yt-dlp is up to date (2021.10.22)
[debug] [generic] Extracting URL: https://drooble.com/song/2858030
[generic] 2858030: Requesting header
WARNING: [generic] Falling back on generic information extractor.
[generic] 2858030: Downloading webpage
[generic] 2858030: Extracting information
ERROR: [generic] Unsupported URL: https://drooble.com/song/2858030
Traceback (most recent call last):
  File "/usr/local/bin/yt-dlp/yt_dlp/extractor/common.py", line 593, in extract
    ie_result = self._real_extract(url)
  File "/usr/local/bin/yt-dlp/yt_dlp/extractor/generic.py", line 3706, in _real_extract
    raise UnsupportedError(url)
yt_dlp.utils.UnsupportedError: Unsupported URL: https://drooble.com/song/2858030
Traceback (most recent call last):
  File "/usr/local/bin/yt-dlp/yt_dlp/extractor/common.py", line 593, in extract
    ie_result = self._real_extract(url)
  File "/usr/local/bin/yt-dlp/yt_dlp/extractor/generic.py", line 3706, in _real_extract
    raise UnsupportedError(url)
yt_dlp.utils.UnsupportedError: Unsupported URL: https://drooble.com/song/2858030

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/bin/yt-dlp/yt_dlp/YoutubeDL.py", line 1280, in wrapper
    return func(self, *args, **kwargs)
  File "/usr/local/bin/yt-dlp/yt_dlp/YoutubeDL.py", line 1305, in __extract_info
    ie_result = ie.extract(url)
  File "/usr/local/bin/yt-dlp/yt_dlp/extractor/common.py", line 610, in extract
    e.msg, video_id=video_id, ie=self.IE_NAME, tb=e.traceback, expected=e.expected, cause=e.cause)
yt_dlp.utils.ExtractorError: [generic] Unsupported URL: https://drooble.com/song/2858030
```

### PR DESCRIPTION
**Pull request by u-spec-png**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [ ] Improvement
- [x] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information
Closes #1527.

### CODE REVIEW

File: yt_dlp/extractor/drooble.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    int_or_none,
+    try_get,
+)
+
+
+class DroobleIE(InfoExtractor):
+    _VALID_URL = r'https?://drooble\.com/(?:(?P<user>[^/]+)?/?(?P<kind>song|videos|music/albums)/(?P<id>\d+)|(?P<id_2>[^/]+)/(?P<kind_2>videos|music))'
+    _TESTS = [{
+        'url': 'https://drooble.com/song/2858030',
+        'md5': 'TODO: md5 sum of the first 10241 bytes of the video file (use --test)',
```
Comment (pukkandan) [inline yt_dlp/extractor/drooble.py:18]: Seems u never actually ran the test :/

File: yt_dlp/extractor/drooble.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+
+class DroobleIE(InfoExtractor):
+    _VALID_URL = r'https?://drooble\.com/(?:(?P<user>[^/]+)?/?(?P<kind>song|videos|music/albums)/(?P<id>\d+)|(?P<id_2>[^/]+)/(?P<kind_2>videos|music))'
+    _TESTS = [{
+        'url': 'https://drooble.com/song/2858030',
+        'md5': '5ffda90f61c7c318dc0c3df4179eb064',
+        'info_dict': {
+            'id': '2858030',
+            'ext': 'unknown_video',
```
Comment (Ashish0804) [inline yt_dlp/extractor/drooble.py:21]: it shouldn't be this lol
U need to manually set ext during extraction ig

Comment (pukkandan) [reply yt_dlp/extractor/drooble.py:21]: it is acceptable only if the site uses multiple formats and there is no way to tell them apart

**Comment by u-spec-png** [comment]
It's ok to merge now.

**Comment by pukkandan** [comment]
Is there any video (not audio) that isn't a youtube embed?

**Comment by u-spec-png** [comment]
> Is there any video (not audio) that isn't a youtube embed?

No, all videos are YouTube embedded.
> Also, I can't find any non-mp3 audio

Can't either it seems like all audio files are mp3.

**Pull request merged**